src/relier/util.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const jose = require('node-jose');
+/**
+ * Relier utility functions
+ * @module relier-util
+ * @private
+ */
+
+/**
+ * @method extractAccessToken
+ * @desc Extracts the `code` parameter from a given redirectUri
+ * @param {string} redirectUri - OAuth redirect URI
+ * @returns {string}
+ */
+function extractAccessToken(redirectUri) {
+  const m = redirectUri.match(/[#?](.*)/);
+
+  if (! m || m.length < 1) return null;
+  const params = new URLSearchParams(m[1].split('#')[0]);
```

Comment:
I don't understand what's going on with the `#` in this function, are there circumstances in which we return parameters in the hash fragment rather than the query string?